blackbox/docs/best_practice/full_restart_upgrade.txt

```diff
@@ -0,0 +1,175 @@
+.. highlight:: sh
+
+.. _full_restart_upgrade:
+
+====================
+Full Restart Upgrade
+====================
+
+The preferred way to upgrade a running CrateDB cluster is via the
+:ref:`rolling_upgrade` method. But this is not always an option (for example,
+when you are upgrading from one minor version to the next).
+
+When a :ref:`rolling_upgrade` is not possible, a full restart upgrade must be
+done if you want to update CrateDB.
+
+.. NOTE::
+
+   A full restart upgrade cannot be done without downtime for your database.
+
+.. CAUTION::
+
+   If you are using Docker, you must not run ``docker service update`` as this
+   performs a :ref:`rolling_upgrade`.
+
+   Instead you must remove the service and then recreate it.
+
+Upgrade Process
+===============
+
+.. WARNING::
+
+    Before upgrading, you should `back up your data`_.
+
+Stop Every Node
+---------------
+
+The first step is to shut down every CrateDB node in your cluster. By this, we
+mean, shut down every instance of the CrateDB daemon. The host operating system
+can be left running.
+
+Manually
+++++++++
+
+To stop the CrateDB daemon manually, run::
+
+    sh$ kill -USR2 `cat /path/to/crate.pid`
+
+Here, replace ``/path/to/crate.pid`` with the path to the PID file you created
+when starting the CrateDB daemon (see :ref:`cli`).
+
+System Packages
++++++++++++++++
+
+If you're using a Debian, Ubuntu, or Red Hat package, you can do::
+
+    sh$ sudo service crate stop
+
+If you're using the Arch Linux package, you can do::
+
+    sh$ sudo systemctl stop crate.service
+
+Upgrade Every Node
+------------------
+
+.. WARNING::
+
+   Make sure that every CrateDB daemon has been stopped before you continue. If
+   you are following this process and you attempt to upgrade your cluster while
+   nodes are still running, you may experience data corruption or data loss.
+
+Once every node has been stopped, you can upgrade CrateDB.
+
+Manually
+++++++++
```

Comment:
i would really consider seperating things like, "how do i stop start restart my cluster" which is totally setup dependent and "what do i need to do when i upgrade"